### PR TITLE
Updating sap_deployer to use naming module

### PR DESCRIPTION
**Pull request by KimForss**
Problem
SAP Deployer is not using the naming module

Solution
Update SAP Deployer to use the naming module

Tests

### CODE REVIEW

File: deploy/terraform/bootstrap/sap_deployer/randomid.tf [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Generates random text for boot diagnostics storage account name
+resource random_id deploy-random-id {
+  byte_length = 4
+}
```
Comment (rukawata) [inline deploy/terraform/bootstrap/sap_deployer/randomid.tf:4]: nit: newline at the end.

File: deploy/terraform/bootstrap/sap_library/variables_local copy.tf [hunk 2]
```diff
@@ -0,0 +1,6 @@
+locals {
+  environment          = lower(try(var.infrastructure.landscape, ""))
+  location             = try(var.infrastructure.region, "")
+  codename             = lower(try(var.infrastructure.codename, ""))
+  management_vnet_name = try(var.infrastructure.vnets.management.name, "MGMT")
+}
```
Comment (rukawata) [inline deploy/terraform/bootstrap/sap_library/variables_local copy.tf:6]: nit: newline.

File: deploy/terraform/bootstrap/sap_library/variables_local copy.tf [hunk 3]
```diff
@@ -0,0 +1,6 @@
+locals {
+  environment          = lower(try(var.infrastructure.landscape, ""))
```
Comment (rukawata) [inline deploy/terraform/bootstrap/sap_library/variables_local copy.tf:2]: now it's called environment.
```suggestion
  environment          = lower(try(var.infrastructure.environment, ""))
```

File: deploy/terraform/run/sap_deployer/randomid.tf [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# Generates random text for boot diagnostics storage account name
+resource random_id deploy-random-id {
+  
+  byte_length = 4
+}
```
Comment (rukawata) [inline deploy/terraform/run/sap_deployer/randomid.tf:5]: nit: newline

File: deploy/terraform/run/sap_deployer/variables_local.tf [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+  //There is no default as the name is mandatory unless arm_id is specified
+  vnet_mgmt_name = local.vnet_mgmt_exists ? split("/", local.vnet_mgmt_arm_id)[8] : try(local.vnet_mgmt.name, "MGMT")
+
+  // Default naming of vnet has multiple parts. Taking the second-last part as the name incase the name ends with -vnet
+  vnet_mgmt_parts     = length(split("-", local.vnet_mgmt_name))
+  vnet_mgmt_name_part = try(substr(upper(local.vnet_mgmt_name), -5, 5), "") == "-VNET" ? substr(split("-", local.vnet_mgmt_name)[(local.vnet_mgmt_parts - 2)], 0, 7) : local.vnet_mgmt_name
+
+
+
```
Comment (rukawata) [inline deploy/terraform/run/sap_deployer/variables_local.tf:20]: nit: too many empty lines.

```suggestion
```

File: deploy/terraform/run/sap_library/variables_local.tf [hunk 6]
```diff
@@ -0,0 +1,6 @@
+locals {
+  environment          = lower(try(var.infrastructure.landscape, ""))
+  location             = try(var.infrastructure.region, "")
+  codename             = lower(try(var.infrastructure.codename, ""))
+  management_vnet_name = try(var.infrastructure.vnets.management.name, "MGMT")
+}
```
Comment (rukawata) [inline deploy/terraform/run/sap_library/variables_local.tf:6]: nit: newline

File: deploy/terraform/terraform-units/modules/sap_deployer/infrastructure.tf [hunk 7]
```diff
@@ -52,7 +47,7 @@ data "azurerm_subnet" "subnet_mgmt" {
 // Creates boot diagnostics storage account for Deployer
 resource "azurerm_storage_account" "deployer" {
   count                     = local.enable_deployers ? 1 : 0
-  name                      = lower(format("%s%s",local.sa_prefix,substr(local.postfix,0,4)))
+  name                      = local.storageaccount_names[0]
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_deployer/infrastructure.tf:50]: see comment in naming module PR #752 , avoid ambiguous index.

File: deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf [hunk 8]
```diff
@@ -3,84 +3,49 @@ Description:
...
   region             = try(var.infrastructure.region, "")
-  environment        = try(var.infrastructure.environment, "")
-  location_short     = try(var.region_mapping[local.region], "unkn")
-  vnet_mgmt_tempname = local.vnet_mgmt.name
-  prefix             = try(var.infrastructure.resource_group.name, upper(format("%s-%s-%s", local.environment, local.location_short, substr(local.vnet_mgmt_tempname, 0, 7))))
-  sa_prefix          = lower(format("%s%s%sdiag", substr(local.environment, 0, 5), local.location_short, substr(local.vnet_mgmt_tempname, 0, 7)))
-  rg_name            = try(var.infrastructure.resource_group.name, format("%s-INFRASTRUCTURE", local.prefix))
+  vnet_mgmt_tempname = try(local.vnet_mgmt.name, "")
+  prefix             = try(var.infrastructure.resource_group.name, var.naming.prefix.DEPLOYER)
+  rg_name            = try(var.infrastructure.resource_group.name, format("%s%s", local.prefix, local.resource_suffixes["deployer-rg"]))
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf:26]: Step into the dict the same way :) 
Same applies to other places - I will not mark all of them.
```suggestion
  rg_name            = try(var.infrastructure.resource_group.name, format("%s%s", local.prefix, local.resource_suffixes.deployer-rg))
```

File: deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf [hunk 9]
```diff
@@ -148,13 +148,13 @@ resource "null_resource" "prepare-deployer" {
...
-      "mkdir -p $HOME/Azure_SAP_Automated_Deployment/WORKSPACES/LOCAL/${azurerm_resource_group.deployer[0].name}",
-      "mkdir $HOME/Azure_SAP_Automated_Deployment/WORKSPACES/SAP_LIBRARY",
-      "mkdir $HOME/Azure_SAP_Automated_Deployment/WORKSPACES/SAP_SYSTEM",
-      "mkdir $HOME/Azure_SAP_Automated_Deployment/WORKSPACES/SAP_LANDSCAPE",
-      "mkdir $HOME/Azure_SAP_Automated_Deployment/WORKSPACES/DEPLOYER",
+      "mkdir -p $HOME/AZURE_SAP_AUTOMATED_DEPLOYMENT/WORKSPACES/LOCAL/${azurerm_resource_group.deployer[0].name}",
+      "mkdir $HOME/AZURE_SAP_AUTOMATED_DEPLOYMENT/WORKSPACES/SAP_LIBRARY",
+      "mkdir $HOME/AZURE_SAP_AUTOMATED_DEPLOYMENT/WORKSPACES/SAP_SYSTEM",
+      "mkdir $HOME/AZURE_SAP_AUTOMATED_DEPLOYMENT/WORKSPACES/SAP_LANDSCAPE",
+      "mkdir $HOME/AZURE_SAP_AUTOMATED_DEPLOYMENT/WORKSPACES/DEPLOYER",
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf:155]: The folder name is camel, please don't change this - if there is a reason to do that, please do that in a separate PR.
Also this will break the current pipelines.

**Review by rukawata — COMMENTED** [review]
Thanks! Most are nit comments.

File: deploy/terraform/bootstrap/sap_library/variables_local copy.tf [hunk 10]
```diff
@@ -0,0 +1,6 @@
+locals {
```
Comment (rukawata) [inline deploy/terraform/bootstrap/sap_library/variables_local copy.tf:1]: why there is a copied file?

File: deploy/terraform/run/sap_deployer/module.tf [hunk 11]
```diff
@@ -10,4 +10,20 @@ module "sap_deployer" {
   options        = var.options
   ssh-timeout    = var.ssh-timeout
   sshkey         = var.sshkey
+  naming         = module.sap_namegenerator.naming
 }
+
+module "sap_namegenerator" {
```
Comment (rukawata) [inline deploy/terraform/run/sap_deployer/module.tf:16]: I noticed that you started to adopt the new style to remove double quotes in other places. If this is the best practice from going forward, please at least make the consistent change in all newly added code.

Also I would appreciate if you could pointe me to the doc with this new terraform guidelines - I will add that to the `CONTRIBUTING.md`.

File: deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf (hunk 9 above)
Comment (KimForss) [reply deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf:155]: OK - reverted

File: deploy/terraform/bootstrap/sap_library/variables_local copy.tf (hunk 10 above)
Comment (KimForss) [reply deploy/terraform/bootstrap/sap_library/variables_local copy.tf:1]: user error :)

File: deploy/terraform/run/sap_deployer/module.tf (hunk 11 above)
Comment (KimForss) [reply deploy/terraform/run/sap_deployer/module.tf:16]: I did not find the official recommendation to do that but I think it is the right way forward- Terraform works with both but to me with a developer background things inside of "" are not types but constants

File: deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf [hunk 12]
```diff
@@ -3,84 +3,49 @@ Description:
...
   // Resource group and location
 
   region             = try(var.infrastructure.region, "")
-  environment        = try(var.infrastructure.environment, "")
-  location_short     = try(var.region_mapping[local.region], "unkn")
-  vnet_mgmt_tempname = local.vnet_mgmt.name
-  prefix             = try(var.infrastructure.resource_group.name, upper(format("%s-%s-%s", local.environment, local.location_short, substr(local.vnet_mgmt_tempname, 0, 7))))
-  sa_prefix          = lower(format("%s%s%sdiag", substr(local.environment, 0, 5), local.location_short, substr(local.vnet_mgmt_tempname, 0, 7)))
-  rg_name            = try(var.infrastructure.resource_group.name, format("%s-INFRASTRUCTURE", local.prefix))
+  vnet_mgmt_tempname = try(local.vnet_mgmt.name, "")
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf:24]: Is this variable still needed? I can't find where it is used.

File: deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf [hunk 13]
```diff
@@ -3,84 +3,49 @@ Description:
...
+  rg_name            = try(var.infrastructure.resource_group.name, format("%s%s", local.prefix, local.resource_suffixes.deployer-rg))
 
   // Management vnet
   vnet_mgmt        = try(var.infrastructure.vnets.management, {})
   vnet_mgmt_arm_id = try(local.vnet_mgmt.arm_id, "")
   vnet_mgmt_exists = length(local.vnet_mgmt_arm_id) > 0 ? true : false
-  vnet_mgmt_name   = local.vnet_mgmt_exists ? split("/", local.vnet_mgmt_arm_id)[8] : format("%s-vnet", local.prefix)
-  vnet_mgmt_addr   = local.vnet_mgmt_exists ? "" : try(local.vnet_mgmt.address_space, "10.0.0.0/24")
+
+  vnet_mgmt_name = local.vnet_mgmt_exists ? split("/", local.vnet_mgmt_arm_id)[8] : format("%s%s", local.prefix, local.resource_suffixes["vnet"])
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf:33]:
```suggestion
  vnet_mgmt_name = local.vnet_mgmt_exists ? split("/", local.vnet_mgmt_arm_id)[8] : format("%s%s", local.prefix, local.resource_suffixes.vnet)
```

File: deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf [hunk 14]
```diff
@@ -3,84 +3,49 @@ Description:
...
+  vnet_mgmt_addr = local.vnet_mgmt_exists ? "" : try(local.vnet_mgmt.address_space, "10.0.0.16/28")
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf:34]: Is there any reason to use a different defaults?
Current defaults comes from issue #644

File: deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf [hunk 15]
```diff
@@ -3,84 +3,49 @@ Description:
...
 
   // Management subnet
   sub_mgmt          = try(local.vnet_mgmt.subnet_mgmt, {})
   sub_mgmt_arm_id   = try(local.sub_mgmt.arm_id, "")
   sub_mgmt_exists   = length(local.sub_mgmt_arm_id) > 0 ? true : false
-  sub_mgmt_name     = local.sub_mgmt_exists ? split("/", local.sub_mgmt_arm_id)[10] : try(local.sub_mgmt.name, format("%s_deployment-subnet", local.prefix))
+  sub_mgmt_name     = local.sub_mgmt_exists ? "" : try(local.sub_mgmt.name, format("%s%s", local.prefix, local.resource_suffixes["deployer-subnet"]))
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf:40]: 1. I think it is okay to give this variable a value even sub_mgmt_exists, instead of "", so logically it is correct.
2. `local.resource_suffixes.deployer-subnet`

File: deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf [hunk 16]
```diff
@@ -3,84 +3,49 @@ Description:
...
   sub_mgmt_prefix   = local.sub_mgmt_exists ? "" : try(local.sub_mgmt.prefix, "10.0.0.16/28")
   sub_mgmt_deployed = try(local.sub_mgmt_exists ? data.azurerm_subnet.subnet_mgmt[0] : azurerm_subnet.subnet_mgmt[0], null)
 
   // Management NSG
   sub_mgmt_nsg             = try(local.sub_mgmt.nsg, {})
   sub_mgmt_nsg_exists      = try(local.sub_mgmt_nsg.is_existing, false)
   sub_mgmt_nsg_arm_id      = local.sub_mgmt_nsg_exists ? try(local.sub_mgmt_nsg.arm_id, "") : ""
-  sub_mgmt_nsg_name        = local.sub_mgmt_nsg_exists ? "" : try(local.sub_mgmt_nsg.name, format("%s_deploymentSubnet-nsg", local.prefix))
+  sub_mgmt_nsg_name        = local.sub_mgmt_nsg_exists ? "" : try(local.sub_mgmt_nsg.name, format("%s%s", local.prefix, local.resource_suffixes["deployer-subnet-nsg"]))
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf:48]: 1. I know the subnet name was not there. To be consistent, can we add it by deriving from arm_id?
2. `local.resource_suffixes.deployer-subnet-nsg`

File: deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf [hunk 17]
```diff
@@ -6,18 +6,21 @@ Description:
...
 */
 
+data azurerm_subscription "primary" {}
+data azurerm_client_config "current" {}
+
 // Public IP addresse and nic for Deployer
 resource "azurerm_public_ip" "deployer" {
   count               = length(local.deployers)
-  name                = format("%s-%s%02d-pip", local.prefix, local.deployers[count.index].name, count.index)
+  name                = length(local.prefix) > 0 ? format("%s_%s%s", local.prefix, local.deployers[count.index].name, local.resource_suffixes["pip"]) : format("%s%s", local.deployers[count.index].name, local.resource_suffixes["pip"])
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf:15]: 1. Is this a real condition where prefix is empty?
2. `local.resource_suffixes.pip`

File: deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf [hunk 18]
```diff
@@ -92,7 +57,7 @@ locals {
   enable_deployers = length(local.deployer_input) > 0 ? true : false
   deployers = [
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf:58]: The problem is not in this PR, but please correct [this](https://github.com/Azure/sap-hana/blob/c74de84b0213c18ee88ec07daddd668ef40f8779/deploy/terraform/terraform-units/modules/sap_namegenerator/vm.tf#L7):
` deployer_vm_names = [for idx in range(var.app_server_count) :`

Deployer counts comes from the length of the deployer list, not app

File: deploy/terraform/bootstrap/sap_deployer/module.tf [hunk 19]
```diff
@@ -4,10 +4,27 @@ Description:
...
+}
+
+module "sap_namegenerator" {
+  source               = "../../terraform-units/modules/sap_namegenerator"
+  environment          = local.environment
+  location             = local.location
+  codename             = local.codename
+  management_vnet_name = local.vnet_mgmt_name_part
+  random_id            = random_id.deploy-random-id.hex
+
```
Comment (rukawata) [inline deploy/terraform/bootstrap/sap_deployer/module.tf:23]: We miss passing the vm count here.

File: deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf [hunk 20]
```diff
@@ -6,18 +6,21 @@ Description:
...
   location            = azurerm_resource_group.deployer[0].location
   resource_group_name = azurerm_resource_group.deployer[0].name
   allocation_method   = "Static"
 }
 
 resource "azurerm_network_interface" "deployer" {
   count               = length(local.deployers)
-  name                = format("%s-%s%02d-nic", local.prefix, local.deployers[count.index].name, count.index)
+  name                = length(local.prefix) > 0 ? format("%s_%s%s", local.prefix, local.deployers[count.index].name, local.resource_suffixes["nic"]) : format("%s%s", local.deployers[count.index].name, local.resource_suffixes["nic"])
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf:23]: same as above.

File: deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf [hunk 21]
```diff
@@ -57,8 +57,8 @@ resource "azurerm_role_assignment" "sub_user_admin" {
 // Linux Virtual Machine for Deployer
 resource "azurerm_linux_virtual_machine" "deployer" {
   count                           = length(local.deployers)
-  name                            = format("%s-%s%02d", local.prefix,local.deployers[count.index].name, count.index)
-  computer_name                   = format("%s%02d", replace(replace(local.deployers[count.index].name,"-",""),"_",""), count.index)
+  name                            = length(local.prefix) > 0 ? format("%s_%s%s", local.prefix, local.deployers[count.index].name, local.resource_suffixes["vm"]) : format("%s%s", local.deployers[count.index].name, local.resource_suffixes["vm"])
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf:60]: same as above.

File: deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf [hunk 22]
```diff
@@ -68,7 +68,7 @@ resource "azurerm_linux_virtual_machine" "deployer" {
   disable_password_authentication = local.deployers[count.index].authentication.type != "password" ? true : false
 
   os_disk {
-    name                 = format("%s-%s%02d-osdisk", local.prefix, local.deployers[count.index].name, count.index)
+    name                 = length(local.prefix) > 0 ? format("%s_%s%s", local.prefix, local.deployers[count.index].name, local.resource_suffixes["osdisk"]) : format("%s%s", local.deployers[count.index].name, local.resource_suffixes["osdisk"])
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf:71]: same as above.

File: deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf [hunk 23]
```diff
@@ -148,13 +148,13 @@ resource "null_resource" "prepare-deployer" {
...
-      "mkdir -p $HOME/Azure_SAP_Automated_Deployment/WORKSPACES/LOCAL/${azurerm_resource_group.deployer[0].name}",
-      "mkdir $HOME/Azure_SAP_Automated_Deployment/WORKSPACES/SAP_LIBRARY",
-      "mkdir $HOME/Azure_SAP_Automated_Deployment/WORKSPACES/SAP_SYSTEM",
-      "mkdir $HOME/Azure_SAP_Automated_Deployment/WORKSPACES/SAP_LANDSCAPE",
-      "mkdir $HOME/Azure_SAP_Automated_Deployment/WORKSPACES/DEPLOYER",
+      "mkdir -p $HOME/azure_sap_automated_deployment/workspaces/local/${azurerm_resource_group.deployer[0].name}",
+      "mkdir $HOME/azure_sap_automated_deployment/workspaces/sap_library",
+      "mkdir $HOME/azure_sap_automated_deployment/workspaces/sap_system",
+      "mkdir $HOME/azure_sap_automated_deployment/workspaces/sap_landscape",
+      "mkdir $HOME/azure_sap_automated_deployment/workspaces/deployer",
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf:155]: Please keep the casing here.

**Review by rukawata — COMMENTED** [review]
Thanks for the PR! Found some issue - ping me if you need clarity!

File: deploy/terraform/bootstrap/sap_deployer/module.tf [hunk 24]
```diff
@@ -4,10 +4,27 @@ Description:
...
+  //These are not needed for the deployer
+  sap_vnet_name = try(var.infrastructure.vnets.sap.name, "")
+  sap_sid       = ""
+  db_sid        = ""
```
[the remark this answers is not in the document]
Comment (KimForss) [reply deploy/terraform/bootstrap/sap_deployer/module.tf:27]: removed

File: deploy/terraform/bootstrap/sap_deployer/module.tf (hunk 19 above)
Comment (KimForss) [reply deploy/terraform/bootstrap/sap_deployer/module.tf:23]: Added a ToDo comment for this as it will require a change in the naming module

File: deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf (hunk 23 above)
Comment (KimForss) [reply deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf:155]: OK

File: deploy/terraform/bootstrap/sap_deployer/module.tf [hunk 25]
```diff
@@ -4,10 +4,22 @@ Description:
...
+
+module "sap_namegenerator" {
+  source               = "../../terraform-units/modules/sap_namegenerator"
+  environment          = local.environment
+  location             = local.location
+  codename             = local.codename
+  management_vnet_name = local.vnet_mgmt_name_part
+  random_id            = random_id.deploy_random_id.hex
+  // ToDo: Once we update the naming module add a deployer_vm_count variable
+  app_server_count     = 1
```
Comment (rukawata) [inline deploy/terraform/bootstrap/sap_deployer/module.tf:24]: I would prefer if this can be correct now in this PR.
We should pass the length of the list of deployers to `sap_namegenerator` as `deployer_vm_count`.

**Review by rukawata — COMMENTED** [review]
Thanks for addressing some of the comments.
I see there are still some unresolved comments. Please check!

File: deploy/terraform/bootstrap/sap_deployer/module.tf (hunk 25 above)
Comment (KimForss) [reply deploy/terraform/bootstrap/sap_deployer/module.tf:24]: fixed

File: deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf (hunk 12 above)
Comment (KimForss) [reply deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf:24]: removed

File: deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf (hunk 14 above)
Comment (KimForss) [reply deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf]: will be addressed in keyvault

File: deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf (hunk 16 above)
Comment (KimForss) [reply deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf]: done

File: deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf [hunk 26]
```diff
@@ -3,84 +3,48 @@ Description:
...
+  rg_name            = try(var.infrastructure.resource_group.name, format("%s%s", local.prefix, local.resource_suffixes.deployer-rg))
 
   // Management vnet
   vnet_mgmt        = try(var.infrastructure.vnets.management, {})
   vnet_mgmt_arm_id = try(local.vnet_mgmt.arm_id, "")
   vnet_mgmt_exists = length(local.vnet_mgmt_arm_id) > 0 ? true : false
-  vnet_mgmt_name   = local.vnet_mgmt_exists ? split("/", local.vnet_mgmt_arm_id)[8] : format("%s-vnet", local.prefix)
-  vnet_mgmt_addr   = local.vnet_mgmt_exists ? "" : try(local.vnet_mgmt.address_space, "10.0.0.0/24")
+  vnet_mgmt_name   = local.vnet_mgmt_exists ? split("/", local.vnet_mgmt_arm_id)[8] : format("%s%s", local.prefix, local.resource_suffixes.vnet)
+  vnet_mgmt_addr   = local.vnet_mgmt_exists ? "" : try(local.vnet_mgmt.address_space, "")
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf:33]: keep the defaults here. Will address the default in separate PRs.

File: deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf (hunk 23 above)
Comment (rukawata) [reply deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf:155]: Sorry, I still see this changed in the PR?

Comment (KimForss) [reply deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf:155]: my bad

File: deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf [hunk 27]
```diff
@@ -3,84 +3,47 @@ Description:
...
   sub_mgmt_deployed = try(local.sub_mgmt_exists ? data.azurerm_subnet.subnet_mgmt[0] : azurerm_subnet.subnet_mgmt[0], null)
 
   // Management NSG
   sub_mgmt_nsg             = try(local.sub_mgmt.nsg, {})
-  sub_mgmt_nsg_exists      = try(local.sub_mgmt_nsg.is_existing, false)
-  sub_mgmt_nsg_arm_id      = local.sub_mgmt_nsg_exists ? try(local.sub_mgmt_nsg.arm_id, "") : ""
-  sub_mgmt_nsg_name        = local.sub_mgmt_nsg_exists ? "" : try(local.sub_mgmt_nsg.name, format("%s_deploymentSubnet-nsg", local.prefix))
+  sub_mgmt_nsg_arm_id      = try(local.sub_mgmt_nsg.arm_id, "")
+  sub_mgmt_nsg_exists      = length(local.sub_mgmt_nsg_arm_id) > 0 ? true : false
+  sub_mgmt_nsg_name        = local.sub_mgmt_nsg_exists ? split("/", local.sub_mgmt_nsg)[8] : try(local.sub_mgmt_nsg.name, format("%s%s", local.prefix, local.resource_suffixes.deployer-subnet-nsg))
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf:46]: According to the naming standard, this should be called:
`U1338-EAUS-deployment-nsg`
not
`U1338-EAUS-MGMT_deploymentSubnet-nsg`
(U1338 is the environment I used in pipeline)

image

File: deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf [hunk 28]
```diff
@@ -6,18 +6,21 @@ Description:
...
+  name                = format("%s_%s%s", local.prefix, local.deployers[count.index].name, local.resource_suffixes.pip) 
   location            = azurerm_resource_group.deployer[0].location
   resource_group_name = azurerm_resource_group.deployer[0].name
   allocation_method   = "Static"
 }
 
 resource "azurerm_network_interface" "deployer" {
   count               = length(local.deployers)
-  name                = format("%s-%s%02d-nic", local.prefix, local.deployers[count.index].name, count.index)
+  name                = format("%s_%s%s", local.prefix, local.deployers[count.index].name, local.resource_suffixes.nic)
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf:23]: nic should not have deployer computer name in it:
I see in pipeline below change will be made this this PR:
`"U1338-EAUS-MGMT-deployer00-nic" -> "U1338-EAUS-MGMT_u1338eausmgmtdeploy00-nic"`

This is what is in the naming spec:
`{ENVIRONMENT}-{REGION_MAP}-{DEPLOY_VNET}_deployment##-nic`

File: deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf [hunk 29]
```diff
@@ -57,8 +57,8 @@ resource "azurerm_role_assignment" "sub_user_admin" {
 // Linux Virtual Machine for Deployer
 resource "azurerm_linux_virtual_machine" "deployer" {
   count                           = length(local.deployers)
-  name                            = format("%s-%s%02d", local.prefix,local.deployers[count.index].name, count.index)
-  computer_name                   = format("%s%02d", replace(replace(local.deployers[count.index].name,"-",""),"_",""), count.index)
+  name                            = format("%s_%s%s", local.prefix, local.deployers[count.index].name, local.resource_suffixes.vm)
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf:60]: `  ~ name                            = "U1338-EAUS-MGMT-deployer00" -> "U1338-EAUS-MGMT_u1338eausmgmtdeploy00" # forces replacement`

Computer name shuold be part of VM name:
according to doc:
image

File: deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf (hunk 27 above)
Comment (KimForss) [reply deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf:46]: updated the naming module to use _deployment-nsg

File: deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf [hunk 30]
```diff
@@ -6,18 +6,21 @@ Description:
...
 */
 
+data azurerm_subscription "primary" {}
+data azurerm_client_config "current" {}
+
 // Public IP addresse and nic for Deployer
 resource "azurerm_public_ip" "deployer" {
   count               = length(local.deployers)
-  name                = format("%s-%s%02d-pip", local.prefix, local.deployers[count.index].name, count.index)
+  name                = format("%s_%s%s", local.prefix, local.deployers[count.index].name, local.resource_suffixes.pip)
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf]: No computer name inside pip name:

`    ~ name                    = "U1338-EAUS-MGMT-deployer00-pip" -> "U1338-EAUS-MGMT_u1338eausmgmtdeploy00-pip" # forces replacement`

Not yet provided in the naming spec though.

**Review by rukawata — COMMENTED** [review]
Some problem with the naming, please check.
They are visible in the log:
https://dev.azure.com/azuresaphana/Azure-SAP-HANA/_build/results?buildId=11338&view=logs&j=3e7c2453-b9e0-5f22-4314-eb3ec9329da1&t=e6a0a175-e92e-585d-f1b1-708558137faa

File: deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf (hunk 29 above)
Comment (KimForss) [reply deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf:60]: updated

File: deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf (hunk 28 above)
Comment (KimForss) [reply deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf:23]: the computername is now part of it

File: deploy/terraform/terraform-units/modules/sap_namegenerator/variables_local.tf [hunk 31]
```diff
@@ -14,18 +14,22 @@ variable codename {
 
 variable management_vnet_name {
   description = "Name of Management vnet"
+  default     = ""
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_namegenerator/variables_local.tf:17]: If the value is required, should not have default.
Otherwise, using empty string as default may cause other problems.
Same applies to the others.

File: deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf [hunk 32]
```diff
@@ -68,7 +68,7 @@ resource "azurerm_linux_virtual_machine" "deployer" {
   disable_password_authentication = local.deployers[count.index].authentication.type != "password" ? true : false
 
   os_disk {
-    name                 = format("%s-%s%02d-osdisk", local.prefix, local.deployers[count.index].name, count.index)
+    name                 = format("%s_%s%s", local.prefix, local.deployers[count.index].name, local.resource_suffixes.osdisk)
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_deployer/vm-deployer.tf:71]: `  ~ name                      = "U1458-EAUS-MGMT-deployer00-osdisk" -> "U1458-EAUS-MGMT_u1458eausmgmtdeploy00-osdisk" # forces replacement
`
Probably change needs to be done in the naming module.
Doc says to use **OsDisk** instead of **osdisk**

**Review by rukawata — COMMENTED** [review]
Most are good, please check the comments.

File: deploy/terraform/terraform-units/modules/sap_namegenerator/variables_local.tf (hunk 31 above)
Comment (KimForss) [reply deploy/terraform/terraform-units/modules/sap_namegenerator/variables_local.tf:17]: The problem is that the management name is required for deployer but not for library or SDU

**Review by rukawata — APPROVED** [review]
lgtm!
